### DISCUSSION TITLE
Cloudflare Worker static assets

### DISCUSSION THREAD

**Discussion by alexturpin** [vikejs/vike-cloudflare#16] [Ideas]
Hey folks,

I've just come back to the Vike ecosystem after a bit, and it's wonderful to see where it is and with things such as Bati and this project here. Wondering if there are any thoughts on making this plugin work with the new Worker static assets announced recently https://developers.cloudflare.com/workers/static-assets/

There are [several advantages](https://developers.cloudflare.com/workers/static-assets/compatibility-matrix/) to using a Worker project instead of a Pages one, including using things like cron and Durable Objects. 

I've been trying to understand how this could work. It seems as if the whole `routes.json` [wouldn't be needed](https://developers.cloudflare.com/workers/static-assets/routing/)? Ideally, in this world, we would have access to the worker code so we could handle things like the `scheduled` event, even if maybe that means giving up control of `fetch` entirely to `vike-cloudflare` so it can do the Hono / HatTip handling itself underneath.

I'd love to hear what others think.

**Comment by magne4000** [comment]
We're planning to support it, please see https://github.com/vikejs/vike/discussions/1901

**Reply by alexturpin** [reply]
Awesome!
